### ISSUE TITLE
Extension <speed> in gpx log file not correct value or units

### ISSUE THREAD

**Issue by ibabbv** [osmandapp/Osmand#7931]
Before entering an issue, please read the information below.

```GitHub is a main development tool for developers. There are hundreds requests a month and there are only few developers. So by opening an issue, please know that your issue will be sent out to all developers and acknowledge that it could be closed without explanation or with a brief message. Comments on the closed issues are also sent to all developers, so you will definitely will be heard. However, there is no guarantee that a developer will pick up the issue to work on it.```

In case you are not sure to open issue or not, please send your request to support support@osmand.net. For **general questions** about OsmAnd, please use the [Google group](https://groups.google.com/forum/#!forum/osmand).

Please be sure to have read our [FAQ](http://osmand.net/help-online) before creating an issue here. (Also, tell us in Google Groups when the FAQ is wrong or outdated.)

The best way to get help about an OsmAnd issue is to create a valid and detailed issue content.
Please give us the following information so that we can try to **reproduce** your issue:

What version of OsmAnd are you using, on what device and Android/iOS version?

Do you use the offline maps offered within the OsmAnd app for download, or online (tile / raster) maps?

If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs, and its edition date.

__*In case of wrong routing:*__

Tell us whether you have used OsmAnd's in-app offline routing, or any online routing provider like YOURS, OpenRouteService or OSRM.

What routing profile is chosen in OsmAnd app? (car, bike, pedestrian, fastest or shortest)

Please specify as exactly as possible the start and end point of your route by telling us city name and street name so that we can find it via in-app address search easily. Also, a permalink from openstreetmap.org can be helpful.

Tell us your expected routing, and how OsmAnd routes, or add screenshots here.

**Comment by ibabbv** [comment]
OSMAND+   3.4.8   Samsung S10e  SM-G970F  Android 9

Hello,
I have some doubts about the speed logging in a gps track (gpx file) by the gps track logger plugin.

Trackpoints look like this:

       <trkpt lat="51.5115702" lon="5.3231929">
        <ele>10.175</ele>
        <time>2019-11-10T13:02:37Z</time>
        <hdop>4</hdop>
        <extensions>
          <speed>1.176</speed>
        </extensions>
      </trkpt>
      <trkpt lat="51.5116141" lon="5.3231132">
        <ele>11.66</ele>
        <time>2019-11-10T13:02:42Z</time>
        <hdop>3</hdop>
        <extensions>
          <speed>1.261</speed>
        </extensions>
      </trkpt>
      <trkpt lat="51.511662" lon="5.3230182">
        <ele>12.315</ele>
        <time>2019-11-10T13:02:47Z</time>
        <hdop>3</hdop>
        <extensions>
          <speed>1.397</speed>
        </extensions>

I assume speed is in m/s. But I know during this hike I was moving most
of the time >6 km/h, so >1.66 m/s.
If I look to the end result, I have walked 21 km in 3hr 20min, and
average speed approx 6.3 km/hr, 1.75 m/s.

But the speed data is hardly hitting 1.5 m/s.

Log file here:
[2019-11-10_14-02_Sun.zip](https://github.com/osmandapp/Osmand/files/3846197/2019-11-10_14-02_Sun.zip)

When I analyse the gpx file with "javawa track analyse", I get the
probably correct speed data:

Trackpoint Date/time Lat.
[°] Lng.
[°] Distance
[m] Time Elevation
[m] Speed
[km/h] Δ Distance
[m] Δ Time
[s] Δ Elevation
[m]
1 2019-11-10 14:02:37 51.5116 5.3232 0 00:00:00 10.2 **5.3** 7 5 1.5
2 2019-11-10 14:02:42 51.5116 5.3231 7 00:00:05 11.7 **6.1** 8 5 0.7
3 2019-11-10 14:02:47 51.5117 5.3230 16 00:00:10 12.3 **6.5** 9 5 0.7
4 2019-11-10 14:02:52 51.5117 5.3229 25 00:00:15 13.0 **5.6** 9 6 0.6
5 2019-11-10 14:02:58 51.5118 5.3228 34 00:00:21 13.6 **6.1** 10 6
0.8
6 2019-11-10 14:03:04 51.5118 5.3227 44 00:00:27 14.4 **5.9** 10 6
0.1
7 2019-11-10 14:03:10 51.5119 5.3226 54 00:00:33 14.5 **5.8** 10 6
-0.6
8 2019-11-10 14:03:16 51.5119 5.3225 64 00:00:39 13.9 **5.5** 9 6
-0.8

Any idea what is wrong with "speed" data in gpx file, (or my
perception)       Any explanation ???

Thanks in advance

Gus

**Comment by sonora** [comment]
I just ran a plausibility test with a file I had recorded hiking this April (so not sure if we introduced a bug in OsmAnd ever since). Sampling rate was 30 sec (in some sections 1 min due to a power management issue on my device).

I extracted the speed values (yes, GPX requires m/s) from the OsmAnd-recorded file (we do not do much post-processing other than routing, so what you see is what the GPS chipset provided), see attached. The average is 1.63, roughly 6km/h, makes sense. But if you look at the distribution, the range is from 0.25 to 8:
- The terrain was hilly, with narrow arroyos, and on that device GPS was turned off between measurements, which does not really enhance the precision of the measured values.
- I also think that the 'speed' values (just like the elevation values) are secondary, the primary precision is given to the lateral position values
- Quite possibly, the speed value precision also suffers from both the generally low speeds, and the fact that the device was in my cargo pant pocket where its motion was the undulation of my thigh, not really a uniform lateral displacement.

Long story short, please perform the "average" analysis over all your recorded speed values to check if that makes sense. If you cannot, maybe you can share your file here for further analysis.

[GPS-measured-hiking-speed.txt](https://github.com/osmandapp/Osmand/files/3847357/GPS-measured-hiking-speed.txt)

**Comment by ibabbv** [comment]
Hello Hardy,

Here is my recorded file attached.  Analyse by javawa.nl gives 21.25 km 
in 3hr20min  avg speed 6.4 km/hr.

The average of the speed recorded values is 1,22 m/s This doeas not take 
in account that some intervals of recording are longer than the 5 second 
standard setting. If corrected for that, the value would be even 
lower.   (6.3 km/hr is  1.75 m/s)

gr Gus
